File: day1/part1.c
#include <stdio.h>
#include <time.h>
#include <windows.h>

int main()
{
    FILE *file = fopen("input.txt", "r");
    if (file == NULL)
    {
        printf("Could not open file\n");
        return 1;
    }

    LARGE_INTEGER frequency;
    LARGE_INTEGER start, end;
    QueryPerformanceFrequency(&frequency);
    QueryPerformanceCounter(&start);

    int sum = 0;
    char line[256];
    while (fgets(line, sizeof(line), file))
    {
        int first = -1, last = -1;
        for (int i = 0; line[i] != '\0'; i++)
        {
            if (line[i] >= '0' && line[i] <= '9')
            {
                if (first == -1)
                {
                    first = line[i] - '0';
                }
                last = line[i] - '0';
            }
        }
        if (first != -1 && last != -1)
        {
            sum += first * 10 + last;
        }
    }

    printf("%d\n", sum);

    QueryPerformanceCounter(&end);
    long long microseconds = (end.QuadPart - start.QuadPart) * 1000000 / frequency.QuadPart;

    printf("%lld μs\n", microseconds);

    fclose(file);
    return 0;
}

File: day1/part1.java
package day1;

import java.io.*;
import java.nio.file.*;
import java.time.*;

public class part1 {
    public static void main(String[] args) throws IOException {
        Instant start = Instant.now();

        Path path = Paths.get("day1/input.txt");
        BufferedReader reader = Files.newBufferedReader(path);

        int sum = 0;
        String line;
        while ((line = reader.readLine()) != null) {
            int first = -1, last = -1;
            for (char ch : line.toCharArray()) {
                if (Character.isDigit(ch)) {
                    if (first == -1) {
                        first = Character.digit(ch, 10);
                    }
                    last = Character.digit(ch, 10);
                }
            }
            if (first != -1 && last != -1) {
                sum += first * 10 + last;
            }
        }

        System.out.println(sum);

        Duration duration = Duration.between(start, Instant.now());
        System.out.println(duration.toMillis() + " ms");

        reader.close();
    }
}